Building any project using the Project Manager built with target INSTALL fails

**Describe the bug**
It is not possible to build any project via the Project Manager when it is launched from the o3de-install folder (after following the [Pre-built SDK engine guide](https://o3de.org/docs/welcome-guide/setup/setup-from-github/#:~:text=Use%20CMake%20to%20build,target%20INSTALL%20--config%20profile%20--%20/m)) due to the following error:
```
CMake Error at C:/o3de-install/cmake/LYWrappers.cmake:690 (if):
  Maximum recursion depth of 1000 exceeded
Call Stack (most recent call first):
  C:/o3de-install/cmake/SettingsRegistry.cmake:66 (ly_de_alias_target)
  C:/o3de-install/cmake/SettingsRegistry.cmake:72 (ly_get_gem_load_dependencies)
```
Please note that the Project Manager built following the instructions for building the [Source Engine](https://o3de.org/docs/welcome-guide/setup/setup-from-github/#:~:text=The%20following%20example,Editor%20--config%20profile%20--%20/m) does build projects successfully without the mentioned issue.

Please refer to the attached video and logs for more information.

**Steps to reproduce**
Steps to reproduce the behavior:
1. Follow the [Pre-built SDK engine guide](https://o3de.org/docs/welcome-guide/setup/setup-from-github/#:~:text=Use%20CMake%20to%20build,target%20INSTALL%20--config%20profile%20--%20/m).
2. Launch O3DE.exe from the o3de-install/bin/Windows/profile/Default folder.
3. Create a new project.
4. Build the created project.

**Expected behavior**
The project is built without any issues.

**Actual behavior**
The project is not built due to the errors.

**Video**

https://user-images.githubusercontent.com/86953659/141764735-10954811-9ef3-44cf-954a-f7b33a2fc2ce.mp4

**Found in Branch**
Stabilization/2110 (76b7e73)

**Desktop**
Device: PC
OS: Windows
Version: 10
CPU: Intel i5-9600K
GPU: RTX 2070
Memory: 16GB

**Additional context**
[CMakeProjectBuildError.log](https://github.com/o3de/o3de/files/7537837/CMakeProjectBuildError.log)
[CMakeOutput.log](https://github.com/o3de/o3de/files/7537839/CMakeOutput.log)

--- a/Gems/Multiplayer/Code/CMakeLists.txt
+++ b/Gems/Multiplayer/Code/CMakeLists.txt
@@ -117,7 +117,7 @@ if (PAL_TRAIT_BUILD_HOST_TOOLS)
     
     ly_create_alias(NAME Multiplayer.Builders  NAMESPACE Gem TARGETS Gem::Multiplayer.Editor)
     # use the Multiplayer.Editor module in tools like the Editor:  Such tools also get the visual debug view:
-    ly_create_alias(NAME Multiplayer.Tools     NAMESPACE Gem TARGETS Gem::Multiplayer.Editor Gem::Multiplayer.Debug Gem::Multiplayer.Builders)
+    ly_create_alias(NAME Multiplayer.Tools     NAMESPACE Gem TARGETS Gem::Multiplayer.Debug Gem::Multiplayer.Builders)
 endif()
 
 if (PAL_TRAIT_BUILD_TESTS_SUPPORTED)

--- a/Gems/Multiplayer/Code/CMakeLists.txt
+++ b/Gems/Multiplayer/Code/CMakeLists.txt
@@ -86,11 +86,31 @@ ly_create_alias(NAME Multiplayer.Servers NAMESPACE Gem TARGETS Gem::Multiplayer 
 
 if (PAL_TRAIT_BUILD_HOST_TOOLS)
     ly_add_target(
+        NAME Multiplayer.Tools.Static STATIC
+        NAMESPACE Gem
+        FILES_CMAKE
+            multiplayer_tools_files.cmake
+        INCLUDE_DIRECTORIES
+            PRIVATE
+                ${pal_source_dir}
+                AZ::AzNetworking
+                Source
+                .
+            PUBLIC
+                Include
+        BUILD_DEPENDENCIES
+            PUBLIC
+                AZ::AzCore
+                AZ::AzFramework
+                AZ::AzNetworking
+                AZ::AzToolsFramework
+    )
+
+    ly_add_target(
         NAME Multiplayer.Editor GEM_MODULE
         NAMESPACE Gem
         FILES_CMAKE
             multiplayer_editor_shared_files.cmake
-            multiplayer_tools_files.cmake
         COMPILE_DEFINITIONS
             PUBLIC
                 MULTIPLAYER_EDITOR
@@ -112,6 +132,7 @@ if (PAL_TRAIT_BUILD_HOST_TOOLS)
                 Gem::Atom_RPI.Public
                 Gem::Atom_RHI.Reflect
                 Gem::Multiplayer.Static
+                Gem::Multiplayer.Tools.Static
                 Gem::Multiplayer.Builders
     )
     
@@ -158,7 +179,6 @@ if (PAL_TRAIT_BUILD_TESTS_SUPPORTED)
             NAMESPACE Gem
             FILES_CMAKE
                 multiplayer_tools_tests_files.cmake
-                multiplayer_tools_files.cmake
             INCLUDE_DIRECTORIES
                 PRIVATE
                     Tests
@@ -168,10 +188,9 @@ if (PAL_TRAIT_BUILD_TESTS_SUPPORTED)
                 PRIVATE
                     AZ::AzTest
                     AZ::AzTestShared
-                    AZ::AzToolsFramework
                     AZ::AzToolsFrameworkTestCommon
-                    AZ::AzNetworking
                     Gem::Multiplayer.Static
+                    Gem::Multiplayer.Tools.Static
         )
         ly_add_googletest(
             NAME Gem::Multiplayer.Builders.Tests

--- a/Gems/Multiplayer/Code/CMakeLists.txt
+++ b/Gems/Multiplayer/Code/CMakeLists.txt
@@ -86,51 +86,11 @@ ly_create_alias(NAME Multiplayer.Servers NAMESPACE Gem TARGETS Gem::Multiplayer 
 
 if (PAL_TRAIT_BUILD_HOST_TOOLS)
     ly_add_target(
-        NAME Multiplayer.Builders.Static STATIC
-        NAMESPACE Gem
-        FILES_CMAKE
-            multiplayer_tools_files.cmake
-        COMPILE_DEFINITIONS
-            PUBLIC
-                MULTIPLAYER_TOOLS
-        INCLUDE_DIRECTORIES
-            PRIVATE
-                .
-                Source
-                ${pal_source_dir}
-            PUBLIC
-                Include
-        BUILD_DEPENDENCIES
-            PUBLIC
-                AZ::AzToolsFramework
-                Gem::Multiplayer.Static
-    )
-
-    # by naming this target Multiplayer.Builders it ensures that it is loaded
-    # in any pipeline tools (Like Asset Processor, AssetBuilder, etc)
-    ly_add_target(
-        NAME Multiplayer.Builders GEM_MODULE
-        NAMESPACE Gem
-        FILES_CMAKE
-            multiplayer_tools_files.cmake
-        INCLUDE_DIRECTORIES
-            PRIVATE
-                Source
-                .
-            PUBLIC
-                Include
-        BUILD_DEPENDENCIES
-            PRIVATE
-                Gem::Multiplayer.Builders.Static
-        RUNTIME_DEPENDENCIES
-            Gem::Multiplayer.Editor
-    )
-
-    ly_add_target(
         NAME Multiplayer.Editor GEM_MODULE
         NAMESPACE Gem
         FILES_CMAKE
             multiplayer_editor_shared_files.cmake
+            multiplayer_tools_files.cmake
         COMPILE_DEFINITIONS
             PUBLIC
                 MULTIPLAYER_EDITOR
@@ -154,7 +114,8 @@ if (PAL_TRAIT_BUILD_HOST_TOOLS)
                 Gem::Multiplayer.Static
                 Gem::Multiplayer.Builders
     )
-
+    
+    ly_create_alias(NAME Multiplayer.Builders  NAMESPACE Gem TARGETS Gem::Multiplayer.Editor)
     # use the Multiplayer.Editor module in tools like the Editor:  Such tools also get the visual debug view:
     ly_create_alias(NAME Multiplayer.Tools     NAMESPACE Gem TARGETS Gem::Multiplayer.Editor Gem::Multiplayer.Debug Gem::Multiplayer.Builders)
 endif()
@@ -197,6 +158,7 @@ if (PAL_TRAIT_BUILD_TESTS_SUPPORTED)
             NAMESPACE Gem
             FILES_CMAKE
                 multiplayer_tools_tests_files.cmake
+                multiplayer_tools_files.cmake
             INCLUDE_DIRECTORIES
                 PRIVATE
                     Tests
@@ -206,8 +168,10 @@ if (PAL_TRAIT_BUILD_TESTS_SUPPORTED)
                 PRIVATE
                     AZ::AzTest
                     AZ::AzTestShared
+                    AZ::AzToolsFramework
                     AZ::AzToolsFrameworkTestCommon
-                    Gem::Multiplayer.Builders.Static
+                    AZ::AzNetworking
+                    Gem::Multiplayer.Static
         )
         ly_add_googletest(
             NAME Gem::Multiplayer.Builders.Tests

--- a/Gems/Multiplayer/Code/Source/MultiplayerGem.cpp
+++ b/Gems/Multiplayer/Code/Source/MultiplayerGem.cpp
@@ -6,10 +6,9 @@
  *
  */
 
+#include <MultiplayerToolsSystemComponent.h>
 #include <AzNetworking/Framework/NetworkingSystemComponent.h>
 #include <Multiplayer/Components/NetBindComponent.h>
-#include <Multiplayer/Components/NetworkHierarchyChildComponent.h>
-#include <Multiplayer/Components/NetworkHierarchyRootComponent.h>
 #include <Source/MultiplayerGem.h>
 #include <Source/MultiplayerSystemComponent.h>
 #include <Source/AutoGen/AutoComponentTypes.h>
@@ -20,25 +19,32 @@ namespace Multiplayer
     MultiplayerModule::MultiplayerModule()
         : AZ::Module()
     {
-        m_descriptors.insert(m_descriptors.end(), {
-            AzNetworking::NetworkingSystemComponent::CreateDescriptor(),
-            MultiplayerSystemComponent::CreateDescriptor(),
-            NetBindComponent::CreateDescriptor(),
-            NetworkSpawnableHolderComponent::CreateDescriptor(),
-        });
+        m_descriptors.insert(
+            m_descriptors.end(),
+            {
+                AzNetworking::NetworkingSystemComponent::CreateDescriptor(),
+                MultiplayerSystemComponent::CreateDescriptor(),
+                NetBindComponent::CreateDescriptor(),
+                NetworkSpawnableHolderComponent::CreateDescriptor(),
+#ifdef MULTIPLAYER_EDITOR
+                MultiplayerToolsSystemComponent::CreateDescriptor(),
+#endif
+            });
 
         CreateComponentDescriptors(m_descriptors);
     }
 
     AZ::ComponentTypeList MultiplayerModule::GetRequiredSystemComponents() const
     {
-        return AZ::ComponentTypeList
-        {
+        return AZ::ComponentTypeList{
             azrtti_typeid<AzNetworking::NetworkingSystemComponent>(),
             azrtti_typeid<MultiplayerSystemComponent>(),
+#ifdef MULTIPLAYER_EDITOR
+            azrtti_typeid<MultiplayerToolsSystemComponent>(),
+#endif
         };
     }
-}
+} // namespace Multiplayer
 
 #if !defined(MULTIPLAYER_EDITOR)
 AZ_DECLARE_MODULE_CLASS(Gem_Multiplayer, Multiplayer::MultiplayerModule);

--- a/Gems/Multiplayer/Code/Source/MultiplayerToolsSystemComponent.cpp
+++ b/Gems/Multiplayer/Code/Source/MultiplayerToolsSystemComponent.cpp
@@ -6,7 +6,7 @@
  *
  */
 
-#include <MultiplayerToolsModule.h>
+#include <MultiplayerToolsSystemComponent.h>
 #include <Pipeline/NetworkPrefabProcessor.h>
 
 #include <AzCore/Serialization/Json/RegistrationContext.h>
@@ -39,22 +39,4 @@ namespace Multiplayer
     {
         m_didProcessNetPrefabs = didProcessNetPrefabs;
     }
-
-    MultiplayerToolsModule::MultiplayerToolsModule()
-        : AZ::Module()
-    {
-        m_descriptors.insert(m_descriptors.end(), {
-            MultiplayerToolsSystemComponent::CreateDescriptor(),
-        });
-    }
-
-    AZ::ComponentTypeList MultiplayerToolsModule::GetRequiredSystemComponents() const
-    {
-        return AZ::ComponentTypeList
-        {
-            azrtti_typeid<MultiplayerToolsSystemComponent>(),
-        };
-    }
 } // namespace Multiplayer
-
-AZ_DECLARE_MODULE_CLASS(Gem_Multiplayer_Tools, Multiplayer::MultiplayerToolsModule);

--- a/Gems/Multiplayer/Code/Source/MultiplayerToolsSystemComponent.h
+++ b/Gems/Multiplayer/Code/Source/MultiplayerToolsSystemComponent.h
@@ -37,19 +37,5 @@ namespace Multiplayer
 
         bool m_didProcessNetPrefabs = false;
     };
-
-    class MultiplayerToolsModule
-        : public AZ::Module
-    {
-    public:
-
-        AZ_RTTI(MultiplayerToolsModule, "{3F726172-21FC-48FA-8CFA-7D87EBA07E55}", AZ::Module);
-        AZ_CLASS_ALLOCATOR(MultiplayerToolsModule, AZ::SystemAllocator, 0);
-
-        MultiplayerToolsModule();
-        ~MultiplayerToolsModule() override = default;
-
-        AZ::ComponentTypeList GetRequiredSystemComponents() const override;
-    };
 } // namespace Multiplayer
 

--- a/Gems/Multiplayer/Code/multiplayer_tools_files.cmake
+++ b/Gems/Multiplayer/Code/multiplayer_tools_files.cmake
@@ -10,6 +10,6 @@ set(FILES
     Include/Multiplayer/IMultiplayerTools.h
     Source/Pipeline/NetworkPrefabProcessor.cpp
     Source/Pipeline/NetworkPrefabProcessor.h
-    Source/MultiplayerToolsModule.h
-    Source/MultiplayerToolsModule.cpp
+    Source/MultiplayerToolsSystemComponent.cpp
+    Source/MultiplayerToolsSystemComponent.h
 )

